Makefile

```diff
@@ -1,69 +1,70 @@
-SHELL = /usr/bin/env bash
-OS_NAME = $(shell uname -s)
-BUILD_CONFIGURATION = Debug
-BINARIES_PATH = $(shell pwd)/Binaries
-SRC_PATH = $(shell pwd)/src
-BOOTSTRAP_PATH = $(BINARIES_PATH)/Bootstrap
-BUILD_LOG_PATH =
-HOME_DIR = $(shell cd ~ && pwd)
-DOTNET_VERSION = 1.0.1
-DOTNET = $(BINARIES_PATH)/dotnet-cli/dotnet
-TARGET_FX = netcoreapp1.1
+SHELL := /usr/bin/env bash
```

Comment:
`=` is lazy, `:=` is strict.

    foo = hi $(bar)
    bar = one
    $(info $(foo))
    bar = two
    $(info $(foo))

prints "hi one" "hi two"

    foo := hi $(bar)
    bar := one
    $(info $(foo))
    bar := two
    $(info $(foo))

prints "hi " "hi " (empty var).

The first behavior seems incredibly insane, so having the default for most variable assignments be `:=` is good style.

Note it gets weird with recursive variables (e.g. our use of `MSBUILD_ARGS`):

    MSBUILD_ARGS = $(MSBUILD_ARGS) /some:switch

makes things get upset ("`Recursive variable 'MSBUILD_ARGS' references itself (eventually).  Stop.`").